### ISSUE TITLE
After Update Landscape failed to finish downloading.

### ISSUE THREAD

**Issue by gndclouds** [urbit/urbit#5042]
**Describe the bug**
Every few landscape updates the update completes but landscape remains all white. If I run `+trouble` I get 

```bash
[ [%base-hash ~[0v1t.cdv9k.ggpop.o6q78.dl64t.densq.9qqo0.3efv5.aa68o.o13ai.4dii7]]
  [%sour-hash 0v1t.cdv9k.ggpop.o6q78.dl64t.densq.9qqo0.3efv5.aa68o.o13ai.4dii7]
  [%home-hash 0vm.277u6.9ag40.bkbas.qgp4l.2djb3.em4cs.5rjqq.7i7m7.v7l8c.l7fu1]
  [%kids-hash 0vm.277u6.9ag40.bkbas.qgp4l.2djb3.em4cs.5rjqq.7i7m7.v7l8c.l7fu1]
  [%glob-hash ~[[/~btc/js/bundle 0v2.2k8eq.k1uhe.9acu7.emum0.s8gjd %waiting] [/~landscape/js/bundle 0v7.5i82f.ntu87.0gevr.qheif.0iut9 %waiting]]]
  [%our ship=~nopmex-ricmel point='1798467254' life=[~ 1] rift=[~ 0]]
  [%sponsor ship=~malbus point='28342' life=[~ 1] rift=[~ 0]]
  [%dopzod ship=~dopzod point='4608' life=[~ 3] rift=[~ 2]]
  "Compare lifes and rifts to values here:"
  "https://etherscan.io/address/azimuth.eth#readContract"
  "  life - getKeyRevisionNumber"
  "  rift - getContinuityNumber"
  ~
]
```

Since the js/bundle it still waiting I then try `:glob [%kick /'~btc'/js/bundle]`.  This works at first but quickly I get the following crash message.

```bash
> :glob [%kick /'~landscape'/js/bundle]
>=
http: fail (29, 504): I/O error (body; content-length)
glob: thread failed; will retry
%crash
/lib/strandio/hoon:<[416 3].[417 46]>
/lib/strandio/hoon:<[415 3].[417 46]>
/lib/strandio/hoon:<[414 3].[417 46]>
/lib/strandio/hoon:<[413 3].[417 46]>
/lib/strandio/hoon:<[426 3].[426 33]>
/lib/strand/hoon:<[102 21].[102 43]>
/lib/strand/hoon:<[102 14].[102 44]>
/lib/strand/hoon:<[97 5].[103 7]>
/lib/strand/hoon:<[96 5].[103 7]>
/lib/strand/hoon:<[95 5].[103 7]>
/lib/strand/hoon:<[93 5].[103 7]>
/lib/strand/hoon:<[94 7].[94 18]>
/lib/strand/hoon:<[93 5].[103 7]>
/lib/strand/hoon:<[142 20].[142 49]>
/lib/strand/hoon:<[142 7].[164 9]>
/lib/strand/hoon:<[139 7].[164 9]>
/lib/strand/hoon:<[138 7].[164 9]>
/app/spider/hoon:<[427 7].[427 61]>
http: fail (30, 504): I/O error (body; content-length)
glob: thread failed; will retry
%crash
/lib/strandio/hoon:<[416 3].[417 46]>
/lib/strandio/hoon:<[415 3].[417 46]>
/lib/strandio/hoon:<[414 3].[417 46]>
/lib/strandio/hoon:<[413 3].[417 46]>
/lib/strandio/hoon:<[426 3].[426 33]>
/lib/strand/hoon:<[102 21].[102 43]>
/lib/strand/hoon:<[102 14].[102 44]>
/lib/strand/hoon:<[97 5].[103 7]>
/lib/strand/hoon:<[96 5].[103 7]>
/lib/strand/hoon:<[95 5].[103 7]>
/lib/strand/hoon:<[93 5].[103 7]>
/lib/strand/hoon:<[94 7].[94 18]>
/lib/strand/hoon:<[93 5].[103 7]>
/lib/strand/hoon:<[142 20].[142 49]>
/lib/strand/hoon:<[142 7].[164 9]>
/lib/strand/hoon:<[139 7].[164 9]>
/lib/strand/hoon:<[138 7].[164 9]>
/app/spider/hoon:<[427 7].[427 61]>
```

This repeats for a few days until it eventually completes.

When running `+trouble` I get 

**To Reproduce**
I don't know how to reproduce this, I leave urbit running in the background all the time, and am unsure what causes this.

**Expected behavior**
After an update Landscape should finish downloading.

**Screenshots**
What Landscape looks like after the update.
<img width="1754" alt="image" src="https://user-images.githubusercontent.com/1757261/120593856-a9dbc400-c472-11eb-86a6-68468b7bdea0.png">

**Desktop (please complete the following information):**
 - OS: MacOS 11.3.1 (20E241)
 - Browser Safari, Chrome, Opera all reproduce the issue
 - Version 4dii7

**Additional context**
I'm sorry I don't have more togo on, let me know if there is anything else I can try or info I can send. The first time was after the update ecdi6

**Comment by lycurgus** [comment]
I'm not sure if it was the same user, but someone who was having the same repeated problem recently also mentioned that they had a bad internet connection at home. My suspicion is that their connection was flaking out and cutting the download short (thus the error about `content-length`). Success would then have been a matter of happening to get internet stable enough to complete the download in one try...

Hopefully that helps narrow down what might be going on here.

**Comment by gndclouds** [comment]
Ok, I'm spending today to fix this, Let me fix the breach, and then will track down the content-length issue.

**Comment by gndclouds** [comment]
Ok, I have breached my ship and the same issue happens. 😞  Could this be an issue with the port or something? I just checked my connection and I have 400MB up and 500MB down with no packet loss so I think for my case this might be something else.

Should I try changing my port configuration?

**Comment by lycurgus** [comment]
There shouldn't be an impact from port configuration - your ship will try to download the files over HTTPS (from `https://bootstrap.urbit.org`) so as long as the internet, including that domain, is accessible from the computer where your ship is running, it should be fine.

Any firewall in place on your network?

**Comment by gndclouds** [comment]
humm, I just checked and the firewall on the computer is off and the router doesn't appear to be blocking it either. I still get the following when running trouble. It looks like it's still waiting for the landscape to download. After breaching I followed the 'install via cli' guide and it looked like it should just work now? Or did I miss something?

```
[ [%base-hash ~[0v11.vr93l.829hc.bskg9.rk25a.fsfv7.ih3he.malb9.mh3um.4q9je.vljf9]]
  [%sour-hash 0v11.vr93l.829hc.bskg9.rk25a.fsfv7.ih3he.malb9.mh3um.4q9je.vljf9]
  [%home-hash 0v1m.32hqq.h6or0.h92se.2rm24.gd925.8oneg.c7nij.4kp7k.hclum.fh7vh]
  [%kids-hash 0v1m.32hqq.h6or0.h92se.2rm24.gd925.8oneg.c7nij.4kp7k.hclum.fh7vh]
  [%glob-hash ~[[/~btc/js/bundle 0v6.h33vk.c3n1c.4gpkn.cci4c.pnh4r %waiting] [/~landscape/js/bundle 0v7.0bqfl.auvn6.m43ts.414as.hp1j4 %waiting]]]
  [%our ship=~nopmex-ricmel point='1798467254' life=[~ 3] rift=[~ 1]]
  [%sponsor ship=~malbus point='28342' life=[~ 1] rift=[~ 0]]
  [%dopzod ship=~dopzod point='4608' life=[~ 3] rift=[~ 2]]
  "Compare lifes and rifts to values here:"
  "https://etherscan.io/address/azimuth.eth#readContract"
  "  life - getKeyRevisionNumber"
  "  rift - getContinuityNumber"
  ~
]
```

**Comment by gndclouds** [comment]
Some updates! I got things working but some odd things to note.

After breaching I moved my ship files to an older backup computer and landscape downloaded with no issues. Then I moved the files back to daily computer and landscape worked.

The odd thing was macOS 11.4 (the daily computer) then gave me a security warning for launching an unsigned app. After allowing it to run from the security & preferences panel everything started working. I will see if this breaks when it tries to updates.

Originally when running Urbit on my Daily computer I keep getting the TCP RST error when debugging in Wireshark (see attached screenshot)
<img width="864" alt="Screen Shot 2021-06-29 at 11 16 58 AM" src="https://user-images.githubusercontent.com/1757261/123742163-ccc88f00-d8dd-11eb-96b2-e296250d7ec8.png">

**Comment by lycurgus** [comment]
An update from someone else who has encountered a similar issue on a Mac (and who doesn't have a github account, so I'm posting on their behalf):

-----

I experienced the same issues as OP when attempting to launch a comet using a fresh Urbit installation on macOS Big Sur 11.5.1. I also tried using Port to create comets, which encountered the same basic problem -- the ship would be made but Landscape wouldn't load. (In Port an error message would be displayed indicating "There may be something wrong with your remote ship")

I have reliable internet and got the same result regardless if I was on VPN or not. `:glob [%kick /'~landscape'/js/bundle]` and more generally `:glob %kick` yielded the same type of output from Dojo that OP reports in this thread.

A fresh install and comet launch on macOS Catalina 10.15.7 worked, with Landscape was accessible by web browser as expected. So this problem seems specific to Big Sur. I tried disabling Gatekeeper on the macOS 11.5.1 install but this did not help Landscape download. Nor did attempts to copy the Urbit folder w/ comet from the macOS 10.15.7 laptop to the macOS 11.5.1 workstation; as any new comet mining or OTA update on the 11.5.1 workstation suffers a glob thread failure.
